src/main/java/com/google/api/gax/grpc/ApiCallSettingsTyped.java

```diff
@@ -1,40 +1,92 @@
 package com.google.api.gax.grpc;
 
+import com.google.api.gax.core.RetrySettings;
+import com.google.common.collect.ImmutableSet;
+
+import io.grpc.ManagedChannel;
 import io.grpc.MethodDescriptor;
+import io.grpc.Status;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
 
 /**
- * A settings class with generic typing that could used to configrue an API method or create
- * the method callable object, which can be directly operated against an API.
+ * Package-private: Use other concrete settings classes instead of this class from outside of
+ * the package.
+ * A settings class with generic typing which can be used to configure an API method or create
+ * the method callable object. This class can be used as the base class that other concrete
+ * call settings classes inherit from.
  */
 class ApiCallSettingsTyped<RequestT, ResponseT> extends ApiCallSettings {
-  protected MethodDescriptor<RequestT, ResponseT> methodDescriptor;
 
-  protected FutureCallable<RequestT, ResponseT> createFutureCallable() {
-    ClientCallFactory<RequestT, ResponseT> clientCallFactory =
-        new DescriptorClientCallFactory<>(methodDescriptor);
-    return new DirectCallable<>(clientCallFactory);
+  private final MethodDescriptor<RequestT, ResponseT> methodDescriptor;
+
+  public MethodDescriptor<RequestT, ResponseT> getMethodDescriptor() {
+    return methodDescriptor;
   }
 
-  ApiCallSettingsTyped(Builder<RequestT, ResponseT> builder) {
-    super(builder);
-    this.methodDescriptor = builder.getDescriptor();
+  public ApiCallSettingsTyped(Set<Status.Code> retryableCodes,
+                              RetrySettings retrySettings,
+                              MethodDescriptor<RequestT, ResponseT> methodDescriptor) {
+    super(retryableCodes, retrySettings);
+    this.methodDescriptor = methodDescriptor;
   }
 
   public static class Builder<RequestT, ResponseT>
       extends ApiCallSettings.Builder {
-    protected MethodDescriptor<RequestT, ResponseT> grpcMethodDescriptor;
+    private MethodDescriptor<RequestT, ResponseT> grpcMethodDescriptor;
 
     public Builder(MethodDescriptor<RequestT, ResponseT> grpcMethodDescriptor) {
       this.grpcMethodDescriptor = grpcMethodDescriptor;
     }
 
-    public MethodDescriptor<RequestT, ResponseT> getDescriptor() {
+    public MethodDescriptor<RequestT, ResponseT> getMethodDescriptor() {
       return grpcMethodDescriptor;
     }
 
     @Override
     public ApiCallSettingsTyped<RequestT, ResponseT> build() {
-      return new ApiCallSettingsTyped<RequestT, ResponseT>(this);
+      return new ApiCallSettingsTyped<RequestT, ResponseT>(
+          getRetryableCodes(), getRetrySettingsBuilder().build(), grpcMethodDescriptor);
+    }
+  }
+
+  protected FutureCallable<RequestT, ResponseT> createFutureCallable() {
```

Comment:
It isn't necessary to have separate createFutureCallable() and setupCallable() methods. Just have one method, createBase() or something like that, which the child classes can call from create().